swiftui/ChartboostMediationDemo/Ads/Banner/BannerAdView.swift

```diff
@@ -86,10 +86,14 @@ struct BannerAdView: View {
                 isBusy = true
             case .failed(let message, let error):
                 isBusy = false
-                failureMessage = "\(message): \(error.localizedDescription)"
+                failureMessage = "\(message): \(error?.localizedDescription ?? "")"
             default:
                 isBusy = false
             }
         }
     }
+
+    init(placementName: String) {
+        self._controller = StateObject(wrappedValue: BannerAdController(placementName: placementName))
```

Comment:
This isn't a great practice in SwiftUI. A state object is maintained as long as the associated view is alive, but the BannerAdView struct itself might be recreated several times.
Can you instead pass the controller as a dependency?
You can achieve this by removing this custom init and replacing the controller definition by:
```
@ObservedObject let controller: BannerAdController
```

Reply:
Actually nvm, I thought you were going to reuse the same view types and pass different controllers.
But I see you've created a new FullscreenAdView instead.
If you want to go that route you can then leave things the way they were before and have the views instantiate their controllers directly on their definition.

Reply:
Ok, I probably will leave that alone then, but I think I see what you were saying in the first comment.
A design where a single "AdView" type accepted different controller types would have been a good way to structure the app.